modules/programs/obs-studio.nix

```diff
@@ -37,19 +22,27 @@ in {
         '';
       };
 
+      finalPackage = mkOption {
+        type = types.package;
+        visible = false;
+        readOnly = true;
+        description = "Resulting customized OBS Studio package.";
+      };
+
       plugins = mkOption {
         default = [ ];
-        example = literalExample "[ pkgs.obs-linuxbrowser ]";
+        example = literalExample "[ pkgs.obs-studio-plugins.wlrobs ]";
         description = "Optional OBS plugins.";
         type = types.listOf types.package;
       };
     };
   };
 
   config = mkIf cfg.enable {
-    home.packages = [ cfg.package ];
-
-    xdg.configFile."obs-studio/plugins" =
-      mkIf (cfg.plugins != [ ]) { source = mkPluginEnv cfg.plugins; };
+    home.packages = [ cfg.finalPackage ];
+    programs.obs-studio.finalPackage =
+      pkgs.wrapOBS.override { obs-studio = cfg.package; } {
```

Comment:
Quick question: if it's only used in this one place and can't be modified nor seen, why not just inline it into the `home.packages = [ ... ];`?

Reply:
I'll leave that to someone more familiar with the `home-manager` repo

`finalPackage` isn't used anywhere else for the `neovim` module either

Reply:
The `finalPackage` pattern is really nice to use when something is wrapped.
This pattern is used with Emacs, Neovim and others.

Reply:
What's it useful for? The usual way to do this (if a non-anonymous binding is desired, which is generally only done if using the binding in multiple places) is with `let`:

```nix
let
  cfg = options.programs.obs-studio;
  finalPackage = wrapOBS.override { obs-studio = cfg.package; } {
    inherit (cfg) plugins;
  };
in {
  /* ... */
  config = mkIf cfg.enable {
    home.packages = [ finalPackage ];
  };
}
```

Being read-only, this isn't possible to set from anywhere else, even using `mkForce`:

```
nix-repl> with lib; (evalModules {
            modules = [
              {
                options.foo = mkOption {
                  type = types.int;
                  readOnly = true;
                };
                config.foo = 1;
              }
              {
                foo = lib.mkForce 2;
              }
            ];
          }).config.foo
error: The option `foo' is read-only, but it's set multiple times. Definition values:
- In `<unknown-file>': 2
- In `<unknown-file>': 1
```

If this is to make the package available from *outside* the module, it wouldn't be marked hidden: `visible = false` implies it's internal to the module/an implementation detail, and shouldn't be considered stable for external consumers. Are you sure this isn't just something that's been copy/pasted between modules a couple of times?